src/pages/_app.tsx

```diff
@@ -22,18 +23,28 @@ const App = ({ Component, pageProps, router }: AppProps): JSX.Element => {
 const AppPage: FC<AppProps> = ({ Component, pageProps, router }) => {
   const [opened, setOpened] = useState(false);
   const [password, setPassword] = useState<string | null>(null);
+  const { currentUser } = useCurrentUser();
 
   useEffect(() => {
     setPassword(localStorage.getItem(TECH_UNI));
     setOpened(true);
-  }, [opened]);
+  }, []);
 
   if (!password) return <AuthModal opened={opened} setOpened={setOpened} />;
   if (router.pathname === LINKS.LOGIN) return <Component {...pageProps} />;
-  // if (currentUser?.status === 0 && router.pathname !== LINKS.SIGNUP && router.pathname !== LINKS.LOGIN) {
+  // if (currentUser?.status !== 2 && router.pathname !== LINKS.SIGNUP && router.pathname !== LINKS.LOGIN) {
   //   return <h1>承認待ちです。</h1>;
   // }
 
+  const isNotAdminUser = currentUser?.position !== 2;
+  const isAdminPage = router.pathname === LINKS.ADMIN;
+  const isAdminIdPage = router.pathname === LINKS.ADMINID;
+  if (isNotAdminUser && (isAdminPage || isAdminIdPage)) {
+    router.push(LINKS.HOME);
+    // TODO: ここをtoast表示させるようにしたいが今の状態だとレンダリン回数的に4つ表示させるようになってしまう。
+    console.log("管理者しか見れません");
+  }
+
```

Comment:
```suggestion
  if (currentUser && isLoading) {
    const isNotAdminUser = currentUser?.position <= 1;
    const isAdminPage = router.pathname === LINKS.ADMIN;
    const isAdminIdPage = router.pathname === LINKS.ADMINID;
    if (isNotAdminUser && (isAdminPage || isAdminIdPage)) {
      router.push(LINKS.HOME);
      // TODO: ここをtoast表示させるようにしたいが今の状態だとレンダリン回数的に4つ表示させるようになってしまう。
      console.log("管理者しか見れません");
    }
```

- 2以上であれば、管理者なので、見れるようにしたい
- ＋で動作確認してみたら、currentUserがnullの時にこの処理が走っていて2であってもhomeに飛ばされている
ので
これをAuthProvider.tsxに移して、suggestionみたいな感じにすればいいのでは？？（一応動作確認してみたらいけたよ！！）

Reply:
> ＋で動作確認してみたら、currentUserがnullの時にこの処理が走っていて2であってもhomeに飛ばされている
> ので

あー確かに俺の前の条件だとその挙動になるね。

> これをAuthProvider.tsxに移して、suggestionみたいな感じにすればいいのでは？？（一応動作確認してみたらいけたよ！！）

なるほど！けどこれ条件のisLoadingは必要...？？
currentUserを塞げばいいだけならここいらない気もするけど...

Reply:
isLoadingだけでいい！

Reply:
あ、ごめん逆だった。
isLoadingいらない！の方！

Reply:
おけおけ！

Reply:
[未ログイン状態でadminページに行った後の挙動の修正](https://github.com/TechUni2020/member-site/pull/83/commits/904ce43634b51425d5c3d94c5e3c72f1ceccdae7) ここで修正しました！！

それから、[ログアウト時に、ローカルストレージのUIDも削除するように](https://github.com/TechUni2020/member-site/pull/83/commits/ee1eb439f5a717f5dd21c9ff084aefcfca64136c)っていうのも追加で行いました！